gdgpisausermanager.py

```diff
@@ -1,138 +1,191 @@
-            user = "@" + update.callback_query.from_user.username
-        except Exception as _:
-            user = update.callback_query.from_user.first_name
+        bot.send_message(
+            chat_id,
+            text="{}: grazie per aver completato la registrazione!".format(user_handle),
+        )
 
-        bot.send_message(update.callback_query.message.chat_id,
-                         text=user + ", grazie per aver completato la registrazione!")
+        logger.info("Utente confermato: {}.".format(user_handle))
 
 
 def check_activity(bot, update):
+    """
+    :type bot: telegram.Bot
+    :type update: telegram.Update
+    """
     bot.send_message(
         chat_id=update.message.chat_id,
         text="pong",
-        reply_to_message_id=update.message.message_id)
+        reply_to_message_id=update.message.message_id,
+    )
 
 
-def is_admin(bot, userid, chatid):
-    return userid in [admin.user.id for admin in bot.get_chat_administrators(chatid)]
+def is_admin(bot, user_id, chat_id):
+    """
+    :type bot: telegram.Bot
+    :type user_id: str or int
+    :type chat_id: str or int
+    """
+    try:
+        return any(user_id == admin.user.id for admin in bot.get_chat_administrators(chat_id))
+    except NetworkError:
+        return False
 
 
 def ban_user(bot, update):
+    """
+    :type bot: telegram.Bot
+    :type update: telegram.Update
+    """
     if update.message.reply_to_message is None:
         return
 
-    if is_admin(bot, update.message.from_user.id, update.message.chat.id):
-        if not is_admin(bot, update.message.reply_to_message.from_user.id, update.message.chat.id):
-            reply_text = "{} è stato bannato con successo.".format(update.message.reply_to_message.from_user.username)
-            bot.kickChatMember(update.message.chat.id, update.message.reply_to_message.from_user.id)
-            bot.send_message(update.message.chat.id, text=reply_text)
+    chat_id = update.message.chat.id
 
-            try:
-                user = "@" + update.message.reply_to_message.from_user.username
-            except Exception as _:
-                user = update.message.reply_to_message.from_user.first_name
+    if is_admin(bot, user_id=update.message.from_user.id, chat_id=chat_id):
+        user_id = update.message.reply_to_message.from_user.id
+        user_handle = update.message.reply_to_message.from_user.name
 
-            logger.info("The user {} was banned with success".format(user))
+        if not is_admin(bot, user_id, chat_id):
+            bot.kick_chat_member(chat_id, user_id)
+            bot.send_message(
+                chat_id,
+                text="{} è stato bannato con successo.".format(user_handle),
+            )
+
+            logger.info("Utente {} bannato con successo.".format(user_handle))
 
 
 def kick_user(bot, update):
+    """
+    :type bot: telegram.Bot
+    :type update: telegram.Update
+    """
+    # FIXME: not clear what is the difference with respect to ban_user
```

Comment:
The difference is that a kicked user could enter again in the group, a banned user can't